backend/src/routes/api/integrations/nim/index.ts

```diff
@@ -61,36 +60,67 @@ module.exports = async (fastify: KubeFastifyInstance) => {
         reply: FastifyReply,
       ) => {
         const enableValues = request.body;
-
-        await createNIMSecret(fastify, namespace, enableValues)
-          .then(async () => {
-            await createNIMAccount(fastify, namespace)
-              .then((response) => {
-                const isEnabled = isAppEnabled(response);
-                reply.send({
-                  isInstalled: true,
-                  isEnabled: isEnabled,
-                  canInstall: false,
-                  error: '',
-                });
-              })
-              .catch((e) => {
-                const message = `Failed to create NIM account, ${e.response?.body?.message}`;
-                fastify.log.error(message);
-                reply.status(e.response.statusCode).send(new Error(message));
+        try {
+          const { createdNew } = await createNIMSecret(fastify, enableValues);
+          if (createdNew) {
+            try {
+              const response = await createNIMAccount(fastify);
+              const isEnabled = isAppEnabled(response);
+              reply.send({
+                isInstalled: true,
+                isEnabled: isEnabled,
+                canInstall: false,
+                error: '',
+              });
+            } catch (accountError: any) {
+              if (accountError.response?.statusCode === 409) {
+                fastify.log.error(`NIM account already exists, skipping creation.`);
+                reply.status(409).send(new Error(`NIM account already exists, skipping creation.`));
+              } else {
+                fastify.log.error(
+                  `Failed to create NIM account. ${accountError.response?.body?.message}`,
+                );
+                reply
+                  .status(accountError.response?.statusCode || 500)
+                  .send(
+                    new Error(
+                      `Failed to create NIM account, ${accountError.response?.body?.message}`,
+                    ),
+                  );
+              }
+            }
+          } else {
+            // Secret was updated, so we skip creating the NIM account but still need to get the status.
```

Comment:
One last thing, so if the `NIMSecret` already exists, we will skip creating the `NIMAccount.` Could there be a scenario were the `NIMSecret` exists, but the `NIMAccount` does not? In this case this code would never create the account. Like if there was a failure and the user is trying to redo it, or the account got deleted accidentally.

Just in case, I think it may be more defensive to also do a create/get for the account no matter if the secret was already existing or not?

Reply:
It is currently possible, but it should no longer be possible once the backend is updated to delete it along with all the associated resources. Should I go ahead and update the code anyway?